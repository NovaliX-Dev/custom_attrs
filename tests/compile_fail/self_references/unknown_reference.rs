use custom_attrs::CustomAttrs;

#[derive(CustomAttrs)]
#[attr(pub a: usize)]
enum Enum {
    #[attr(a = #self.5)]
    Variant1(usize, u32),

    #[attr(a = #self.unknown)]
    Variant2 {
        #[allow(unused)]
        field1: u32,

        #[allow(unused)]
        field2: usize,
    },
}

fn main() {}
