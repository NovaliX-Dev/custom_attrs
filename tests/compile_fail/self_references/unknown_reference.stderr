error[E0425]: cannot find value `_5` in this scope
 --> tests/compile_fail/self_references/unknown_reference.rs:6:22
  |
6 |     #[attr(a = #self.5)]
  |                      ^ help: a local variable with a similar name exists: `_0`

error[E0425]: cannot find value `unknown` in this scope
 --> tests/compile_fail/self_references/unknown_reference.rs:9:22
  |
9 |     #[attr(a = #self.unknown)]
  |                      ^^^^^^^ not found in this scope
